automation_tools/satellite6/capsule.py

```diff
@@ -0,0 +1,398 @@
+"""Tasks for helping automating the provisioning of Satellite 6 Capsules"""
+from __future__ import print_function
+import json
+
+from fabric.api import env, get, put, run, settings, task
+from fabric.operations import _AttributeList
+
+
+class _AttributeDict(dict):
+    """Simple dict subclass to allow arbitrary attibute access"""
+
+
+class ImproperlyConfigured(Exception):
+    """Indicates that the capsule configuration is somehow improrpely configure
+    - for example, if the configuration is not loaded or some required
+    configuration is missing.
+
+    """
+
+
+class Credentials(object):
+    """Stores a server SSH credentials information.
+
+    Usage::
+
+        cred1 = Credentials('admin', 'password')
+        cred2 = Credentials(key_filename='/path/to/ssh.key')
+
+    """
+    def __init__(self, user=None, password=None, key_filename=None):
+        self.user = user
+        self.password = password
+        self.key_filename = key_filename
+
+
+class HostConfig(Credentials):
+    """Stores a host's hostname and credentials information.
+
+    Usage::
+
+        cred1 = HostConfig('host1.example.com', 'admin', 'password')
+        cred2 = HostConfig(
+            'host2.example.com', key_filename='/path/to/ssh.key')
+
+    """
+    def __init__(self, hostname=None, port=22, *args, **kwargs):
+        super(HostConfig, self).__init__(*args, **kwargs)
+        self.hostname = hostname
+        self.port = port
+
+    @property
```

Comment:
Looks reasonable. Is it correct to think that setters or other additional logic may be implemented in the future?